src/persistentstorage.cpp

```diff
@@ -258,3 +266,229 @@ void MysqlPersistentStorage::writelog(const std::string &id, const ObjectValue &
+        curl_easy_setopt(curl, CURLOPT_WRITEDATA, &result);
+
+		res = curl_easy_perform(curl);
+		if (res == CURLE_OK)
+        {
+            long response_code;
+            curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &response_code);
+            if (response_code / 100 == 2)
+                ret = true;
+            else
+                logger_m.errorStream() << "request to " << uri_m <<  " returned HTTP " << response_code << endlog;
+        }
+        else
+        {
+			logger_m.errorStream() << "request to " << uri_m <<  " failed: " << curl_easy_strerror(res) << endlog;
+        }
+#if INFLUXDB_EXTRA_DEBUG
+        logger_m.debugStream() << "response from db is: " << result << endlog;
+#endif
+		curl_easy_cleanup(curl);
+	}
+    return ret;
+}
+
+void InfluxdbPersistentStorage::exportXml(ticpp::Element* pConfig)
+{
+    pConfig->SetAttribute("type", "influxdb");
+    pConfig->SetAttribute("uri", uri_m);
+    pConfig->SetAttribute("user", user_m);
+    pConfig->SetAttribute("pass", pass_m);
+    pConfig->SetAttribute("db", db_m);
+    pConfig->SetAttribute("persist_db", persist_db_m);
+}
+
+void InfluxdbPersistentStorage::write(const std::string& id, const std::string& value)
+{
+    logger_m.infoStream() << "Persisting '" << value << "' for object '" << id << "'" << endlog;
+
+    std::stringstream query_s(std::ios_base::out|std::ios_base::binary);;
+    std::string resp;
+
+    if (value.find(':'))
+    {   // it's necessary to put TimeObjectValue in quotes
+        query_s << id << " " << "val=\"" << value << "\" 0";
+    }
+    else
+    {
+        query_s << id << " " << "val=" << value << " 0";
+    }
+
+    curlRequest(INFLUXDB_WRITE, persist_db_m, query_s.str(), resp);
+}
+
+std::string InfluxdbPersistentStorage::read(const std::string& id, const std::string& defval)
+{
+    std::string value = defval, resp;
+    std::stringstream query_s;
+    query_s << "q=SELECT val FROM \"" << id << "\"";
+
+    if (curlRequest(INFLUXDB_QUERY, persist_db_m, query_s.str(), resp))
+    {
+        Json::CharReaderBuilder builder;
+        Json::CharReader * reader = builder.newCharReader();
+        Json::Value root, val;
+        std::string errors;
+
+        if(!reader->parse(resp.c_str(), resp.c_str()+resp.size(), &root, &errors))
+        {
+            logger_m.warnStream() << "couldn't parse JSON! Errors: " << errors << endlog;
+        }
+        else
+        {
+            val = root["results"][0]["series"][0]["values"][0][1];
+            if (val != Json::nullValue)
+            {
+                value = val.asString();
+                logger_m.debugStream() << "Read '" << value << "' for object '" << id << "' from InfluxDB response " << resp << endlog;
+                return val.asString();
+            }
+            else
+            {
+                logger_m.warnStream() << "id '" << id << "' not found in persistence response " << resp << endlog;
+            }
+        }
+    }
+    
+    return defval;
```

Comment:
At this point, `defval` is always equal to the empty string. No need to keep that local var, then.

Reply:
I am still seing the unused `defval` variable. Did you maybe forget to commit or push something?

Reply:
`defval` is the default value given to the function as an argument which is returned in the error case of not being able to retrieve the data from the database, equivalent to the API of the correpsonding `read` functions of `FilePersistentStorage` and `MysqlPersistentStorage` or am i wrong?

Reply:
@fraxinas No, you are right. I overlooked the fact that `defval` is a parameter of the method. I don't see any use cases for this parameter in the codebase so we may want to get rid of it anyway. But as for your PR, it is fine this way. Sorry for the confusion.